src/ImageProcessor.Web/HttpModules/ImageProcessingModule.cs

```diff
@@ -826,7 +826,10 @@ private IImageService GetImageServiceForRequest(string url, string applicationPa
             foreach (IImageService service in services)
             {
                 string key = service.Prefix;
-                if (!string.IsNullOrWhiteSpace(key) && path.StartsWith(key, StringComparison.InvariantCultureIgnoreCase))
+                if (!string.IsNullOrWhiteSpace(key) 
+                    && (service.IsRegex 
```

Comment:
Yep happy to make this change.. as long as you don't see any problems with existing prefix values containing any regex special characters ? Regex special characters are `[\^$.|?*+()`
I would think the dot (.) character is the most risky of these?

E.g. requests like `http://www.domain.com/remote-axd?...` would be matched with by the RemoteImageService: `<service prefix="remote.axd" name="RemoteImageService" ...>`